### ISSUE TITLE
Wildcards and DuckDNS support

### ISSUE THREAD

**Issue by thelamer** [linuxserver/docker-letsencrypt#132]
DuckDNS now supports TXT records which means you can get a wildcard cert issued for any subdomain of your subdomain. 

The process is below: 

1. Request the wildcard cert: 

```
./certbot-auto certonly --manual -d *.yourdomain.duckdns.org --preferred-challenges dns-01 --server https://acme-v02.api.letsencrypt.org/directory
```

2. Enter the TXT record via the duckdns api

```
curl https://www.duckdns.org/update?domains=yourdomain.duckdns.org\&token=<yourtoken>\&txt=<yourtxt>
```

**Comment by hugepants** [comment]
+1
Support for the DuckDNS API for dns-01 validation would be great.

**Comment by aptalca** [comment]
I have been thinking about this for a while. I do agree that duckdns would be a pretty good free dns provider to support, however the amount of work it would require is not trivial.

Currently, we rely on the official certbot plugins for dns validation: https://github.com/certbot/certbot
The advantage is that we provide certbot with the dns plugin's ini and everything is automated.

To implement duckdns (there is no plugin for it) we would have to essentially create a plugin for it (or put in all the additional logic with getting the key from acme, update on duckdns, and let acme know it's up, etc.). It would also add quite a few more if/else statements to the already too convoluted cert generation process.

In the meantime, I continue to recommend cloudflare as the ideal free dns provider for certbot validation (service is free but once still needs to pay for a domain name).

**Comment by thelamer** [comment]
Given it is just a Curl endpoint you can use a custom hook for renewal requests IE I use this for my txt api endpoint: 

```
  # Attempt a cert issue
  /usr/local/bin/certbot-auto certonly \
  --verbose \
  --noninteractive \
  --manual \
  --server "https://acme-v02.api.letsencrypt.org/directory" \
  --preferred-challenges dns \
  --agree-tos \
  --manual-public-ip-logging-ok \
  --email="${EMAIL}" \
  -d "*.${SERVERIP}" \
--manual-auth-hook /usr/local/bin/txthook
```

With the hook just being a bash script that sleeps: 

```
#! /bin/bash

# Hit the Taisun TXT update endpoint with the txt record from certbot
curl --silent https://api.taisun.io/txtupdate?key=${DNSKEY}\&txt=${CERTBOT_VALIDATION} > /dev/null

# Sleep to ensure deployment
sleep 120
```

**Comment by aptalca** [comment]
Did a quick test on this. I don't think you can cover both `*.subdomain.duckdns.org` and `subdomain.duckdns.org` with one cert. It's one or the other. Will look into it more.

Something looks wrong, though. With manual dns validation with acme requires you to enter both the wildcard and the base url as parameters, and certbot prints the following:
```
Performing the following challenges:
dns-01 challenge for xxx.duckdns.org
dns-01 challenge for xxx.duckdns.org
```
notice the two addresses are the same, and then it expects you to set 2 txt records for the same address, which of course results in an invalid txt record for one of them.

I'm really confused about how one is supposed to get a cert via manual plugin to cover both wildcard subdomains and the main domain.

**Comment by aptalca** [comment]
Here's the rest of it, makes no sense

```
-------------------------------------------------------------------------------
Please deploy a DNS TXT record under the name
_acme-challenge.xxx.duckdns.org with the following value:

PPv2Wom0EufmoNsV0Tol_xxxxxxxxxxxxxx

Before continuing, verify the record is deployed.
-------------------------------------------------------------------------------
Press Enter to Continue

-------------------------------------------------------------------------------
Please deploy a DNS TXT record under the name
_acme-challenge.xxx.duckdns.org with the following value:

tExvr9Su7JPGuT9e_2iZnX9q-xxxxxxxxxxxxxxxxx-s

Before continuing, verify the record is deployed.
-------------------------------------------------------------------------------
Press Enter to Continue
```

**Comment by aptalca** [comment]
Well, whaddya know. The manual method is buggy, but if you use the auth hook, it does the validation of the two domains separately, with the txt setting steps in between, so it works.

I'll go ahead and add wildcard duckdns certs in a bit

**Comment by aptalca** [comment]
Well, the auth hook also sets the txt records first, and then validates them altogether, which doesn't work with duckdns because it only allows one txt record per domain, and the second one overwrites the first.

Here's an issue with certbot: https://github.com/certbot/certbot/issues/6496

**Comment by homerr** [comment]
Done & merged in #232
